paasta_tools/marathon_serviceinit.py

```diff
@@ -220,21 +220,20 @@ def status_marathon_job_verbose(service, instance, client, cluster, soa_dir):
     # For verbose mode, we want to see *any* matching app. As it may
     # not be the one that we think should be deployed. For example
     # during a bounce we want to see the old and new ones.
-    for app_id in marathon_tools.get_matching_appids(service, instance, client):
-        if marathon_tools.is_app_id_running(app_id, client):
-            app = client.get_app(app_id)
-            tasks, output = get_verbose_status_of_marathon_app(
-                marathon_client=client,
-                app=app,
-                service=service,
-                instance=instance,
-                cluster=cluster,
-                soa_dir=soa_dir,
-            )
-            all_tasks.extend(tasks)
-            all_output.append(output)
-        else:
-            all_output.append("Warning: App %s is not running yet." % app_id)
+
+    relevant_clients = clients.get_all_clients_for_service(job_config)
+    marathon_apps_with_clients = marathon_tools.get_marathon_apps_with_clients(relevant_clients)
+    for app, client in marathon_tools.get_matching_apps_with_clients(service, instance, marathon_apps_with_clients):
+        tasks, output = get_verbose_status_of_marathon_app(
+            marathon_client=client,
+            app=app,
+            service=service,
+            instance=instance,
+            cluster=cluster,
+            soa_dir=soa_dir,
+        )
+        all_tasks.extend(tasks)
+        all_output.append(output)
```

Comment:
Can you keep the previous behavior with something like
```
if not `marathon_tools.get_matching_apps_with_clients`:
  append(Warning...)
```
but save the variable so you don't have to call the function twice

Reply:
So, I removed this because `get_matching_appids` calls `get_all_marathon_apps` and returns a subset of the results; then `is_app_id_running` makes sure the app id is in the results of `list_all_marathon_app_ids`, which calls `get_all_marathon_apps`. There's no real way for the else block to fire. It's effectively like writing:

```
for thing in things:
    if thing in things:
        ...
    else:
        raise Exception(...)
```